### DISCUSSION TITLE
Typescript interpreting DocumentType<Class> as any

### DISCUSSION THREAD

**Discussion by SaadGillani** [typegoose/typegoose#701] [Q&A]
I have a function with the following signature
`public isAdmin(user: DocumentType<User>): boolean {}`
but typescript is considering the argument user to be of type any 

class user looks like following:
```ts
class User {
  @prop({ required: true })
  _id: string;

  @prop({ required: false })
  email?: string;

  @prop({ required: true, validate: nameRegex })
  firstName: string;

  @prop({ required: true, validate: nameRegex })
  lastName: string;
  }
```

**Comment by hasezoey** [comment]
When opening a issue please use a issue template (unless it is absolutely not fitting), and not remove the template

because this seems more like a question, i will move to discussions

---

please provide more information, like versions of: nodejs, typescript, typegoose, mongoose, editor (like vscode)

**Reply by adeelahmad96** [reply]
@hasezoey 
**Versions**

NodeJS: v16.14.2

Typescript: 4.6.3

Typegoose: 8.2.0

mongoose: 5.13.9

editor: vscode

**Reply by hasezoey** [reply]
maybe consider upgrading to the latest typegoose and mongoose version, aside from that:
- do you maybe still have `@types/mongoose` installed? when yes, remove it
- did you already try to reinstall all dependencies? (clear `node_modules`?)
- do normal mongoose types work (not any)?
- do you maybe have mongoose installed multiple times? (unlikely) (`yarn why mongoose`, `npm ls mongoose`)
